### ISSUE TITLE
P2-BE-13 - quant/payoff/max_loss.py — max loss and breakevens for multi-leg structures.

### ISSUE THREAD

**Issue by Riwan000** [Riwan000/Alpaca_hackathon#64]
**Task `P2-BE-13`** - Phase 2 (Deterministic quant engine) - track: `backend`

**Owner:** this workspace (backend/DB/ops)

## Task

`quant/payoff/max_loss.py` — max loss and breakevens for multi-leg structures.

## Detail / Test / Confirm

- Test: `tests/quant/test_max_loss.py` (P2-BE-21) — put spread max loss = net debit; collar bounded both sides; breakevens correct.
- [ ] Confirm — `pytest tests/quant/test_max_loss.py -q` green.

---
Source: `docs/phased-implementation-plan.md`. Done = code + its **Test** green + the **Confirm** box checked here.

**Comment by Riwan000** [comment]
Done in `27f9b92` (code + tests), plan checkbox in `7c7527c`.

`backend/quant/payoff/max_loss.py` -- pure, dependency-free. The combined payoff is piecewise-linear with a kink at every strike, so its extrema sit at a strike, at spot 0, or out at infinity:
- `structure_risk(legs)` -> `StructureRisk(max_loss, max_profit, worst_pnl, best_pnl, breakevens)`
- `max_loss` is a positive magnitude (0.0 if the structure cannot lose), `inf` when the right tail falls away (e.g. naked short call)
- `max_profit` is `inf` when the right tail rises without bound
- `breakevens` -- ascending zero-crossings, from strike-kink interpolation plus the analytic right-tail piece
- `tail_slopes(legs)` -> (left, right) gradients; `max_loss` / `max_profit` / `breakevens` convenience wrappers

Test: `tests/quant/test_max_loss.py` -- 10 cases. Bear put debit spread: max loss = net debit (4), max profit = width - debit (6), breakeven 96. Collar: bounded both sides (loss 5 / profit 10), breakeven 100. Naked short call: max loss = inf, breakeven strike + premium. Long call: max profit = inf.

    pytest tests/quant/test_max_loss.py -q  ->  10 passed

Full quant suite green (375 passed); black-clean.
